wxGUI location wizard: First page

lindakladivova opened the pull request:
Location wizard location name marked as required and the Project Location renamed to Location Name.

> @lindakladivova Please provide screenshots before and after. It's self-explanatory description :-)

Sorry I do not have before. I have the one after :-) :
![Screenshot from 2020-05-18 15-28-09](https://user-images.githubusercontent.com/49241681/82257184-388baa80-991d-11ea-8fc2-20fdf30dd7bd.png)




> I would say the asterisk should go right after the label. Since there is GridBagSizer used, you may need to shift the edit fields one column to the right to get the asterisk there so that everything is aligned. Alternatively, the edit fields could be moved below the label like we have in forms.

Here's the screenshot of the second version:
![Screenshot from 2020-05-19 10-06-21](https://user-images.githubusercontent.com/49241681/82344664-f2d0ef80-99b9-11ea-81d8-bef65cfba872.png)

Here's the screenshot of the third version:
(The layout was a bit changed and also the Description was marked as optional. GIS data directory is still non-editable but made through wx.TE_READONLY method.)
![Screenshot from 2020-05-19 15-04-43](https://user-images.githubusercontent.com/49241681/82373246-edd46600-99e2-11ea-8b23-3b165ee8f4de.png)

Here's the screenshot of the fourth version:
![Screenshot from 2020-05-20 11-08-53](https://user-images.githubusercontent.com/49241681/82470901-8f16f700-9a8b-11ea-8f33-45be60470ca6.png)
I more like the TextCtrl than StaticText for Location Database. I find it more logical because the window is editable. What do you think?

The fifth version with StaticText:
![Screenshot from 2020-05-20 12-09-40](https://user-images.githubusercontent.com/49241681/82476212-4e22e080-9a93-11ea-8662-c4f79d700415.png)

landam commented:
@lindakladivova  Please provide screenshots before and after. It's self-explanatory description :-)

petrasovaa commented on gui/wxpython/location_wizard/wizard.py:
```diff
@@ -179,6 +179,11 @@ def __init__(self, wizard, parent, grassdatabase):
         self.tlocTitle = self.MakeTextCtrl(size=(400, -1))
+        
+        # text for required options
+        self.required_txt = self.MakeStaticText("*", size=(-1, -1))
```
I don't think the required_txt needs to be attribute of the class (remove self).

petrasovaa commented:
We thought about it more and I think we should redo the DB part. The readonly textctrl is I think confusing (visually you can't see it's readonly at least on gtk). I think we can use statictext as before and move it below, it shouldn't be the first thing. Here is a poor drawing of what you could do. Notice also change in page title and Location Name could be just Name. The Change button is just if we realize we need to enable users to change the database (would be aligned with the asterisk).
![IMG_20200519_225401111](https://user-images.githubusercontent.com/7494312/82400168-ac64aa80-9a24-11ea-89ac-c0fdbb26c677.jpg)

I think you could also go ahead and remove the checkboxes (that's in the roadmap).

petrasovaa commented on gui/wxpython/location_wizard/wizard.py:
```diff
@@ -171,14 +162,26 @@ def __init__(self, wizard, parent, grassdatabase):
         self.tlocTitle = self.MakeTextCtrl(size=(400, -1))
+        
+        # text for required options
+        self.required_txt = self.MakeLabel("*")
```
As I mentioned, don't use self here if it's used only locally within the scope of the method, if you make it an attribute of the class it looks like it's being used elsewhere too, which is confusing.

petrasovaa commented on gui/wxpython/location_wizard/wizard.py:
```diff
@@ -171,14 +162,26 @@ def __init__(self, wizard, parent, grassdatabase):
+        self.required_txt.SetToolTip(_("This option is required"))
+        
+        # text for optional options
+        self.optional_txt = self.MakeLabel("(optional)")
```
Make it translatable: _("optional")

petrasovaa commented on gui/wxpython/location_wizard/wizard.py:
```diff
@@ -171,14 +162,26 @@ def __init__(self, wizard, parent, grassdatabase):
+        self.optional_txt = self.MakeLabel("(optional)")
+        italics = wx.Font(10, wx.DEFAULT, wx.ITALIC, wx.NORMAL)
+        self.optional_txt.SetFont(italics)
+        self.optional_txt.SetForegroundColour("gray")
```
I would recommend here using system colors instead of 'gray': `wx.SystemSettings.GetColour(wx.SYS_COLOUR_GRAYTEXT)` because people have different window styles.

petrasovaa commented on gui/wxpython/location_wizard/wizard.py:
```diff
@@ -171,14 +162,26 @@ def __init__(self, wizard, parent, grassdatabase):
         # text controls
-        self.tgisdbase = self.MakeStaticText(grassdatabase, size=(-1, -1))
-        self.tlocation = self.MakeTextCtrl("newLocation", size=(300, -1))
+        self.tgisdbase = self.MakeTextCtrl(grassdatabase, size=(400, -1), style = wx.TE_READONLY | wx.TRANSPARENT_WINDOW)       
```
If you look at the code checks, the flake8 code quality check is failing on your PR because of spaces around =. It should be:
`..., style=wx.TE_READONLY | wx.TRANSPARENT_WINDOW`

see [PEP8](https://www.python.org/dev/peps/pep-0008/) for reference.

lindakladivova commented:
> We thought about it more and I think we should redo the DB part. The readonly textctrl is I think confusing (visually you can't see it's readonly at least on gtk). I think we can use statictext as before and move it below, it shouldn't be the first thing. Here is a poor drawing of what you could do. Notice also change in page title and Location Name could be just Name. The Change button is just if we realize we need to enable users to change the database (would be aligned with the asterisk).
> ![IMG_20200519_225401111](https://user-images.githubusercontent.com/7494312/82400168-ac64aa80-9a24-11ea-89ac-c0fdbb26c677.jpg)
> 
> I think you could also go ahead and remove the checkboxes (that's in the roadmap).

Another version:
I also removed check boxes.
![Screenshot from 2020-05-20 11-08-53](https://user-images.githubusercontent.com/49241681/82469907-47dc3680-9a8a-11ea-8d0e-fc17641fe27b.png)

petrasovaa commented:
> I also removed check boxes.
> ![Screenshot from 2020-05-20 11-08-53](https://user-images.githubusercontent.com/49241681/82469907-47dc3680-9a8a-11ea-8d0e-fc17641fe27b.png)

I think we should use only StaticText for the DB to communicate clearly that user doesn't need to edit it.

lindakladivova commented:
> > I also removed check boxes.
> > ![Screenshot from 2020-05-20 11-08-53](https://user-images.githubusercontent.com/49241681/82469907-47dc3680-9a8a-11ea-8d0e-fc17641fe27b.png)
> 
> I think we should use only StaticText for the DB to communicate clearly that user doesn't need to edit it.

Ok. I have changed that:
![Screenshot from 2020-05-20 12-09-40](https://user-images.githubusercontent.com/49241681/82476057-161b9d80-9a93-11ea-9a9b-27bbdfc8d70d.png)

petrasovaa commented on gui/wxpython/location_wizard/wizard.py:
```diff
@@ -164,58 +155,51 @@ class DatabasePage(TitledPage):
         self.tlocTitle = self.MakeTextCtrl(size=(400, -1))
-
-        # checkbox
-        self.tlocRegion = self.MakeCheckBox(_("Set default region extent and resolution"),
```
Just removing the widgets is not enough, there are probably used somewhere.

petrasovaa commented:
Looks good! Just one other small change, I noticed the summary page still has Location Title, so it would be good to change that to Description to sync the labels.
